app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/KnowledgeManagementPortalSearchController.php

```diff
@@ -67,7 +68,19 @@ public function get(Request $request): KnowledgeManagementSearchData
             KnowledgeBaseItem::query()
                 ->public()
                 ->with('tags')
-                ->when($search->isNotEmpty(), fn (Builder $query) => $query->tap(new SearchBy('title', $search)))
+                ->when(
+                    ArticleFullTextSearch::active() && $search->isNotEmpty(),
+                    fn (Builder $query) => $query
+                        ->whereRaw(
+                            "search_vector @@ websearch_to_tsquery('english', ?)",
+                            [$search]
+                        )
+                        ->orderByRaw(
+                            "ts_rank_cd(search_vector, websearch_to_tsquery('english', ?)) DESC",
+                            [$search]
+                        ),
+                    fn (Builder $query) => $query->tap(new SearchBy('title', $search))
```

Comment:
Setting the old search query to the "default" here is not going to work. As this will now result in the old query being triggered if `$search` is empty.

You need to leave it as two different whens but check the `ArticleFullTextSearch` Feature flag in each.